pinot-common/src/main/java/org/apache/pinot/common/exception/QueryException.java

```diff
@@ -144,17 +153,29 @@ public static ProcessingException getException(ProcessingException processingExc
     return copiedProcessingException;
   }
 
-  public static String getTruncatedStackTrace(Exception exception) {
+  public static String getTruncatedStackTrace(Throwable exception) {
     StringWriter stringWriter = new StringWriter();
     exception.printStackTrace(new PrintWriter(stringWriter));
     String fullStackTrace = stringWriter.toString();
     String[] lines = fullStackTrace.split("\n");
-    int numLinesOfStackTrace = Math.min(lines.length, _maxLinesOfStackTrace);
-    int lengthOfStackTrace = numLinesOfStackTrace - 1;
-    for (int i = 0; i < numLinesOfStackTrace; i++) {
-      lengthOfStackTrace += lines[i].length();
+    // exception should at least have one line, no need to check here.
+    StringBuilder sb = new StringBuilder(lines[0]);
+    int lengthOfStackTrace = 1;
```

Comment:
(nit)
```suggestion
    int numLinesOfStackTrace = 1;
```

Reply:
technically not the same thing. i specifically dont want to name them the same since the _maxNumLineOfStackTrace actually was meant for _maxNumLineOfStackTrace per frame. I will adjust the whole naming